architecture/adr-list/adr-002-identity-transactions.md

```diff
@@ -0,0 +1,272 @@
+
+**Example:**
+
+```javascript
+{
+  "id": "N22KY2Dyvmuu2PyyqSFKue#key-0",
+  "type": "JsonWebKey2020",
+  "controller": "N22KY2Dyvmuu2PyyqSFKue",
+  "publicKeyJwk": {
+    "kty": "OKP", // external (property name)
+    "crv": "Ed25519", // external (property name)
+    "x": "VCpo2LMLhn6iWku8MKvSLg2ZAoC-nlOyPVQaO3FxVeQ" // external (property name)
+}
+```
+
+#### Service
+
+1. **`id`** (string): The value of the id property MUST be a URI conforming to [RFC3986](https://www.rfc-editor.org/rfc/rfc3986). A conforming producer MUST NOT produce multiple service entries with the same ID. A conforming consumer MUST produce an error if it detects multiple service entries with the same ID. It has a follow format: `<DIDDoc-id>#<service-alias>`
+2. **`type`** (string): The service type and its associated properties SHOULD be registered in the DID Specification Registries [DID-SPEC-REGISTRIES](https://www.w3.org/TR/did-spec-registries/)
+3. **`serviceEndpoint`** (strings): A string that conforms to the rules of [RFC3986](https://www.rfc-editor.org/rfc/rfc3986) for URIs, a map, or a set composed of a one or more strings that conform to the rules of [RFC3986](https://www.rfc-editor.org/rfc/rfc3986) for URIs and/or maps.
+
+**Example:**
+
+ ```javascript
+"service": [{
+  "id":"N22KY2Dyvmuu2PyyqSFKue#linked-domain",
+  "type": "LinkedDomains",
+  "serviceEndpoint": "https://bar.example.com"
+}]
+```
+
+#### Update `DID`
+
+If there is no DID transaction with the specified DID \(`dest`\), it is considered as a creation request for a new DID.
+
+If there is a DID transaction with the specified DID \(`dest`\), then this is update of existing DID. In this case, we can specify only the values we would like to override. All unspecified values remain the same. E.g., if a key rotation needs to be performed, the owner of the DID needs to send a DID transaction request with `dest`, `verkey` only. `alias` will stay the same.
+
+**Note**: Fields `dest` and `owner` should have the same value.
+
+#### State format
+
+`id -> {encode(data, creators), tx_hash, tx_timestamp }`
+
+### `SCHEMA`
+
+This transaction is used to create a Schema associated with credentials.
+
+It is not possible to update an existing Schema, to ensure the original schema used to issue any credentials in the past are always available.
+
+If a Schema evolves, a new schema with a new version or name needs to be created.
+
+- **`data`**: Dictionary with Schema's data:
+  - **`id`**: DID as base58-encoded string for 16 or 32 byte DID value.
+  - **`attr_names`**: Array of attribute name strings (125 attributes maximum)
+  - **`name`**: Schema's name string
+  - **`version`**: Schema's version string
+
+#### `SCHEMA` transaction format
+
+```javascript
+{
+  "id": "N22KY2Dyvmuu2PyyqSFKue",
+  "version": "1.0",
+  "name": "Degree",
+  "attr_names": ["undergrad", "last_name", "first_name", "birth_date", "postgrad", "expiry_date"]
+}
+```
```

Comment:
This should be stored as a DIDDoc, and as defined here it's not stored in a DIDDoc compliant fashion. The format must therefore be changed.

One consequence therefore is much like a DIDDoc therefore, it has the same fields pretty much as other DIDocs (verification method, controllers, etc) and can be updated. Question is: can the older versions be fetched?

Reply:
Brent gave us the idea that schema and cred_def are resources, DDDoc simply describes a specific identifier.
We discussed this idea a lot earlier within the architectural team. Sergey K (as an Indy architect) and I do not recommend turning the schema and cred_def into a DiDDoc. These entities have a different class and a different unique set of fields for each resource.

And this file describes a ledger format of transaction. Theoretically, we can transform it on a client side

Reply:
I guess this is a broader conversation with Richard perhaps but the rest of the SSI ecosystem is moving towards DIDDoc to use this, so how much do we wanna future proof?